### PR TITLE
Snackbar on profile screen after registering pregnancy

### PR DESCRIPTION
**Pull request by kelvin-ngure**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

#2980 

**Engineer Checklist**
- [ ] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [ ] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [ ] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [ ] I have built and run the FHIRCore app to verify my change fixes the issue and/or does not break the app 
- [ ] I have checked that this PR does NOT introduce **breaking changes** that require an update to **_Content_** and/or **_Configs_**? _If it does add a sample here or a link to exactly what changes need to be made to the content._

**Code Reviewer Checklist**
- [ ] I have verified **Unit tests** have been written for any new feature(s) and edge cases
- [ ] I have verified any strings visible on UI components are in the `strings.xml` file
- [ ] I have verifed the [CHANGELOG.md](./CHANGELOG.md) file has any notable changes to the codebase
- [ ] I have verified the solution has been implemented in a configurable and generic way for reuseable components
- [ ] I have built and run the FHIRCore app to verify the change fixes the issue and/or does not break the app

### CODE REVIEW

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/profile/ProfileFragment.kt [hunk 1]
```diff
@@ -83,14 +91,40 @@ class ProfileFragment : Fragment() {
...
+        LaunchedEffect(Unit) {
+          profileViewModel.snackBarStateFlow.hookSnackBar(
+              scaffoldState = scaffoldState,
+              resourceData = null,
+              navController = findNavController(),
           )
         }
+
+        AppTheme {
+          Scaffold(
```
Comment (ellykits) [inline android/quest/src/main/java/org/smartregister/fhircore/quest/ui/profile/ProfileFragment.kt:106]: The profile screen already has a scaffold. I do not think this is needed/necessary. The ProfileViewModel already includes a snackbar flow.

Comment (ellykits) [reply android/quest/src/main/java/org/smartregister/fhircore/quest/ui/profile/ProfileFragment.kt:106]: Perhaps the issue to investigate here is why the Snackbar is not appearing upon submission of a Questionnaire.

Comment (kelvin-ngure) [reply android/quest/src/main/java/org/smartregister/fhircore/quest/ui/profile/ProfileFragment.kt:106]: Yes, I just realized that while investigating. I've stopped adding code and I'm still exploring the flow. I apologize for that

**Comment by ellykits** [comment]
The implementation provided in this PR already exists; closing this.

**Pull request closed without merging**